File: eojin_project_sql/insert_user.c
// gcc insert_user.c -o insert_user -lmysqlclient

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <mysql/mysql.h>

int main() {
    MYSQL *conn;
    const char *server = "localhost";
    const char *user = "bangme";
    const char *password = "djwls123";
    const char *database = "study_db";

    conn = mysql_init(NULL);
    if (conn == NULL) {
        fprintf(stderr, "mysql_init() 실패\n");
        return EXIT_FAILURE;
    }

    if (mysql_real_connect(conn, server, user, password, database, 0, NULL, 0) == NULL) {
        fprintf(stderr, "mysql_real_connect() 실패: %s\n", mysql_error(conn));
        mysql_close(conn);
        return EXIT_FAILURE;
    }

    char user_id[20];
    char name[50];
    char contact[20];

    printf("▶ 사용자 ID를 입력하세요 (문자열): ");
    fgets(user_id, sizeof(user_id), stdin);
    user_id[strcspn(user_id, "\n")] = '\0';

    printf("▶ 이름을 입력하세요: ");
    fgets(name, sizeof(name), stdin);
    name[strcspn(name, "\n")] = '\0';

    printf("▶ 연락처를 입력하세요: ");
    fgets(contact, sizeof(contact), stdin);
    contact[strcspn(contact, "\n")] = '\0';

    char query[256];
    snprintf(query, sizeof(query),
             "INSERT INTO users (user_id, name, contact) VALUES ('%s', '%s', '%s')",
             user_id, name, contact);

    if (mysql_query(conn, query)) {
        fprintf(stderr, "❌ 사용자 INSERT 실패: %s\n", mysql_error(conn));
        mysql_close(conn);
        return EXIT_FAILURE;
    }

    printf("✅ 사용자 등록 완료!\n");

    mysql_close(conn);
    return EXIT_SUCCESS;
}

File: database_module/insert_user.py
# insert_user.py

import mysql.connector

# DB 연결 정보
db_config = {
    "host": "localhost",
    "user": "bangme",
    "password": "djwls123",
    "database": "study_db"
}

def main():
    try:
        conn = mysql.connector.connect(**db_config)
        cursor = conn.cursor()

        user_id = input("▶ 사용자 ID를 입력하세요 (문자열): ").strip()
        name = input("▶ 이름을 입력하세요: ").strip()
        contact = input("▶ 연락처를 입력하세요: ").strip()

        query = """
            INSERT INTO users (user_id, name, contact)
            VALUES (%s, %s, %s)
        """
        cursor.execute(query, (user_id, name, contact))
        conn.commit()

        print("✅ 사용자 등록 완료!")

    except mysql.connector.Error as err:
        print(f"❌ MySQL 오류: {err}")

    finally:
        if cursor:
            cursor.close()
        if conn:
            conn.close()

if __name__ == "__main__":
    main()
